### ISSUE TITLE
[EXT4] [Linux] Sync client assuming I renamed a folder after deleting but it's a new one

### ISSUE THREAD

**Issue by PVince81** [owncloud/client#4588]
### Steps
1. Setup client against OC 8.2.3 (that's where I tested), also on a slower machine (mine has a HDD)
2. Create a folder "pics"
3. Put a few pics inside, for example "pics/pic1.jpg", "pics/pic2.jpg", etc
4. Wait for sync
5. Delete "pics"
6. Quickly after that, create a new folder "pics2"
### Expected result

Desktop client sends DELETE "pics"
### Actual result

Desktop client sends "MOVE pics pics2" then deletes every file one by one instead of the whole folder.

The nasty part is that now instead of having a single folder "pics" in the trashbin, I have all files

Reproduced two times already.
### Versions

ownCloud server v8.2.3 (git) with trashbin enabled
owncloud-client-2.1.1-1.1.x86_64

@guruz @ogoffart

**Comment by dragotin** [comment]
yes, the reason is that the file system on your client recycles inodes. We know the cause, saw it only on Linux so far and thus accepted it as "not nice, but nothing dramatic happens". This conclusion was not assuming the trashbin problem tough. 

Maybe if we use checksums we can improve that?

**Comment by PVince81** [comment]
Another idea would be to use extended attributes to set the ownCloud's fileid on the local files, if the FS supports it. We had a similar idea for the server: https://github.com/owncloud/core/issues/23392 (but unrelated to this issue)

**Comment by PVince81** [comment]
@dragotin I don't think checksums will help, because in my test case it's a folder that was renamed, not a file. :frowning:

**Comment by moscicki** [comment]
Just an idea: maybe use directory mtime/ctime to help identify this case?

**Comment by dragotin** [comment]
@PVince81 I was thinking about something like: The checksum above the checksums of the contents of a directory.

@moscicki yes, but we will need milliseconds resolution, as this command shows that seconds are not enough:
```
stat test && rm -rf test && mkdir test && stat test
```
Hmm, in my case, the inode always changes. @PVince81 which filesystem do you use?

**Comment by PVince81** [comment]
@dragotin it was ext4 on a computer with a slow HDD

**Comment by dragotin** [comment]
about ext4: http://comments.gmane.org/gmane.comp.file-systems.ext4/39671

**Comment by moscicki** [comment]
Maybe you could/should cross-corelate the state-based information (inode)
with events issued by the fs (when available).  It would possibly allow you
to detect the situation in which you recently have seen rm on the given
path and conclude that the inode has been reused (and thus should be
forgotten)?

On Mon, Mar 21, 2016 at 11:46 AM, Klaas Freitag <notifications@github.com>
wrote:

> about ext4: http://comments.gmane.org/gmane.comp.file-systems.ext4/39671
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/owncloud/client/issues/4588#issuecomment-199222287>
>



-- 
---
Best regards,
Kuba

**Comment by ogoffart** [comment]
I think we could check if there is at least one file that has not changed in the folder.
